v2/jwk/jwk.go

```diff
@@ -82,34 +81,36 @@ func (ks JWKeySet) GetPublicKey() (_ *rsa.PublicKey, err error) {
 
 	decodedN, err := base64.RawURLEncoding.DecodeString(ks.Mod)
 	if err != nil {
-		err = errors.Wrap(err, "failed to decode key set `mod`")
-		return
+		return nil, fmt.Errorf("failed to decode key set `mod`: %w", err)
 	}
 
 	pubKey.N.SetBytes(decodedN)
 
 	return pubKey, nil
 }
 
-func GetKeySets() (_ JWKeySets, err error) {
+func GetKeySets() (JWKeySets, error) {
+	getKeySetsLock.Lock()
+	defer getKeySetsLock.Unlock()
```

Comment:
Why was this mutex added?

Reply:
RefreshKeys does a call towards the SSO API (the `/jwks` endpoint). I figured since this is done in multi-threaded contexts the lock prevents this being done multiple times.

Reply:
Makes sense, thanks for clarifying! :+1: